### PR TITLE
fix: reintegrate PasswordStrength (#4111)

### PR DESCRIPTION
**Pull request by singhh-piyush**
- This fix issue: #4111

## PR Description:

Removes `PasswordStrength` enum and associated logic from `lib/models/users.py`. This is dead code following the refactor in `88b91ae`.

### Changes:
* **`lib/models/users.py`**: Deleted `PasswordStrength` class and methods (`value`, `color`, `strength`, `_check_password_strength`).
* **Imports**: Removed `Enum`, `auto`, and `tr` from `users.py`.
* **`lib/models/__init__.py`**: Removed `PasswordStrength` from imports and `__all__`.

## Tests and Checks
- [x] I have tested the code!

### Verification:
* **Tests**: `pytest tests/` passed (13/13).
* **Linting**: Passed `ruff`, `mypy`, `flake8`, and `pylint`.
* **Audit**: `grep -r "PasswordStrength" archinstall/` returned 0 results.
* **Style**: Tab indentation preserved.

### CODE REVIEW

**Comment by singhh-piyush** [comment]
- Fixes issue: #4111

## PR Description
Restores the `PasswordStrength` validation logic that was removed during the commit `88b91ae`. Based on feedback, instead of removing the dead code I integrated it into the password prompt flow.

### Changes:
* **`archinstall/lib/menu/util.py`**: Imported `PasswordStrength` and `warn` (leveraging the existing `tr` import). Integrated the `strength()` check immediately after the initial password input. It now emits a localized CLI warning for `VERY_WEAK`, `WEAK`, and `MODERATE` passwords before proceeding to the confirmation prompt.
* **`tests/test_password_strength.py`**: Added a new test suite with 7 cases covering all enum tiers and edge cases (empty strings, digit-only, etc.).

## Tests and Checks
- [x] I have tested the code locally.

### Verification:
* **Tests**: `pytest tests/` passed (20/20, 0 regressions).
* **Linting**: Passed `ruff`, `mypy`, `flake8`, and `pylint` via local pre-commit hooks.
* **Style**: Tab indentation enforced and verified.

**Comment by singhh-piyush** [comment]
Logic and tests for `PasswordStrength` integrated. I've reverted all locale changes to keep this PR focused on the requested fix .

**Technical Note:** The `translation-check` CI is currently failing on `master` independent of this PR. A duplicate `msgid` in `locales/hi/LC_MESSAGES/base.po` (line ~1491) causes `locales_generator.sh` to abort mid-run, leaving the committed locale files out of sync with the generator's output. I've verified this reproduces on a clean checkout of `master` with no other changes.

All logic has been verified locally with `pytest` - 20/20 passed, including 7 new cases covering every `PasswordStrength` tier and edge cases.

File: archinstall/lib/menu/util.py [hunk 1]
```diff
@@ -33,6 +34,13 @@ def get_password(
 					continue
 
 		password = Password(plaintext=result.get_value())
+
+		strength = PasswordStrength.strength(password.plaintext)
+		if strength in (PasswordStrength.VERY_WEAK, PasswordStrength.WEAK):
+			warn(tr('Weak password. Use uppercase, lowercase, numbers, and symbols.'))
```
Comment (svartkanin) [inline archinstall/lib/menu/util.py]: I'm not sure ther is much point to this. These warnings will only appear in the logs nothing else. 
If the intention is to give feedback to the user regarding their password this will need to be done via the `Input` widget which. There's a validate atribute which gets triggered on `Enter` but nothing that just displays a simple info based on the chars entered, so that will have to be added

**Comment by singhh-piyush** [comment]
<html><head></head><body><p>Thanks for the feedback! I integrated <code>PasswordStrength</code> into the password prompt as a non-blocking hint. The hint renders beneath the input field and updates on each keystroke with color based on strength tier.</p>
<ul>
<li><code>VERY_WEAK</code> / <code>WEAK</code> - red</li>
Screenshot_20260307_162118

<li><code>MODERATE</code> - yellow</li>
Screenshot_20260307_162155

<li><code>STRONG</code> - green</li>
Screenshot_20260307_162212

</ul>
<p><strong>Files changed</strong></p>
<ul>
<li><code>util.py</code> - <code>password_hint</code> closure, wired to first <code>Input()</code> call only</li>
<li><code>components.py</code> - <code>#input-info</code> label, <code>on_input_changed</code> handler, CSS classes</li>
<li><code>helpers.py</code> - type hint updated to <code>tuple[str, str] | None</code></li>
</ul>
<p><strong>Tests</strong></p>
<p>Tested locally in a QEMU VM running the Arch ISO. Screenshots attached.</p>

Password | Result
-- | --
abc | Too weak - too short (red)
abcdefg | Weak - add uppercase, lowercase, numbers and symbols (red)
Abcdefgh12345 | Moderate - increase length (yellow)
Abcdefgh12345!@# | Strong (green)


<p><code>pytest tests/</code> - 20/20 passed. <code>mypy</code>, <code>ruff</code>, <code>flake8</code>, <code>pylint</code> - clean.</p>
<p><strong>

**Comment by CelestifyX** [comment]
My personal opinion is that adding this functionality if it forces you to make a more complex password makes no sense at all, but if it’s just a warning and then it will skip this password, then that’s a different matter.

**Comment by singhh-piyush** [comment]
> My personal opinion is that adding this functionality if it forces you to make a more complex password makes no sense at all, but if it’s just a warning and then it will skip this password, then that’s a different matter.

Hey, it's a non-blocking warning only. You can still proceed with any password regardless of strength, more of a hint/warning for the user.

i added this to address issue #4111 where `PasswordStrength` became dead code after `get_password()` was moved in commit `88b91ae`. Rather than deleting the enum, this restores its original intent as a user-facing hint.

**Comment by singhh-piyush** [comment]
hey @svartkanin , you were right, `warn()` was bypassing the TUI entirely and only hitting the logs. Thanks for the feedback, I really appreciate it 

I made changes as suggested. Extended the `Input` widget with an `info_callback`. Feedback now renders below the field as the user enters a password, with no interference with the `validate` attribute.

Could you take a look at the PR when you get a chance?

Solves issue #4111

**Comment by svartkanin** [comment]
I'll try getting around to test this this week

File: archinstall/lib/menu/util.py [hunk 2]
```diff
@@ -12,12 +12,26 @@ def get_password(
 	preset: str | None = None,
 	skip_confirmation: bool = False,
 ) -> Password | None:
+	def password_hint(value: str) -> tuple[str, str] | None:
+		if not value:
+			return None
+		strength = PasswordStrength.strength(value)
+		if strength == PasswordStrength.VERY_WEAK:
+			return (tr('Too weak - too short'), 'input-hint-weak')
```
Comment (svartkanin) [inline archinstall/lib/menu/util.py]: Can we reformulate these messages as they seem quite strongly worded, I'd also image that users know why they are weak so I'm not sure we need to add context (e.g. Add uppercase) to it either.

I'd rename these to 
* `Password strength: Weak`
* `Password strength: Moderate` 
* `Password strength: Strong`

File: archinstall/tui/ui/components.py [hunk 3]
```diff
@@ -727,6 +727,22 @@ class InputScreen(BaseScreen[str]):
 		color: red;
 		text-align: center;
 	}
+
+	#input-info {
+		text-align: center;
+	}
+
+	.input-hint-weak {
```
Comment (svartkanin) [inline archinstall/tui/ui/components.py]: You are introducing a specific CSS type into the generic input component that is only relevant for password strength.
These should be kept generic 

```
	.input-hint-msg-error {
		color: red;
	}

	.input-hint-msg-warning {
		color: yellow;
	}

	.input-hint-msg-info {
		color: green;
	}
```

File: archinstall/tui/ui/components.py [hunk 4]
```diff
@@ -738,6 +754,7 @@ def __init__(
 		allow_reset: bool = False,
 		allow_skip: bool = False,
 		validator: Validator | None = None,
+		info_callback: Callable[[str], tuple[str, str] | None] | None = None,
```
Comment (svartkanin) [inline archinstall/tui/ui/components.py]: Lets create a data class and enum to handle these better

```
class InputInfoType(Enum):
    MsgInfo = auto()
    MsgWarning = auto()
    MsgError = auto()

@dataclass
class InputInfo:
    message: str
    info_type: InputInfoType
```

**Comment by singhh-piyush** [comment]
Thank you for the feedback. Sorry for the messy commit history. I was working on my local archinstall at the same time, testing some BTRFS formatting changes and some of those unrelated changes got added into this branch by accident. The PR is now clean.

All three points from the review addressed:

- Simplified to Password strength: `Weak / Moderate / Strong` as suggested.

- renamed to generic `.input-hint-msg-error`, `.input-hint-msg-warning`, `.input-hint-msg-info`.

- replaced `tuple[str, str]` with `InputInfo` dataclass and `InputInfoType` enum as suggested.

---
Regarding `VERY_WEAK` - it is already defined in the existing `PasswordStrength` enum in `users.py` and is returned by the `strength()` classifier. I used it in `password_hint` to handle all four possible return values from that function. On the UI side, `VERY_WEAK` and `WEAK` are collapsed into the same display string `'Password strength: Weak'`, so the user only sees three levels

Tested locally in a QEMU VM.
Screenshot_20260315_232005
Screenshot_20260315_232037
Screenshot_20260315_232046

**Comment by svartkanin** [comment]
Could you rebase master, there was a bigger change merged which unfortunately conflicted with your branch

**Comment by singhh-piyush** [comment]
> Could you rebase master, there was a bigger change merged which unfortunately conflicted with your branch

Done, rebased on top of master and the are conflicts resolved.

**Review by svartkanin — APPROVED** [review]
